firmware/Core/Src/stm32_internal_flash_drivers/stm32_internal_flash_drivers.c

```diff
@@ -0,0 +1,143 @@
+#include "stm32_internal_flash_drivers/stm32_internal_flash_drivers.h"
+
+#include "stm32l4r5xx.h"
+#include "stm32l4xx_hal.h" // must include this before stm32l4xx_hal_flash.h
+#include "stm32l4xx_hal_flash.h"
+#include "stm32l4xx_hal_flash_ex.h"
+#include <string.h>
+
+/// @brief Writes data to the flash memory in chunks of 8 bytes.
+/// @param address Address in the flash memory where the data will be written.
+/// @param data uint8_t buffer containing the data to be written.
+/// @param length Length of the data to be written.
+/// @return 0 on success, > 0 on error, 10 if HAL_FLASH_Unlock() failed, 11 if HAL_FLASH_Lock() failed
+/// @note Currently, only allowed to write to golden copy region
+/// @note if data is not 8 bytes long, it will write the data given, and clear the rest of the data
```

Comment:
This note doesn't make sense. Are you saying if it's not a multiple of 8? Less than 8? More than 8? Your example shows that it clears the data "after", not "before", I think. Maybe an example written out as an array would be more helpful.

Reply:
does the following make sense:

```
/// @note Writes data in chunks of 8 bytes.
/// Ex: Suppose we wanted to write to address 0x00, and suppose that at address 0x00, the first 8 bytes looks like the following:
/// [1,2,3,4,5,6,7,8]. If we wanted to write [25,26,27,28], it would result in the following: [25,26,27,28,0,0,0,0], clearing the rest of the bytes.
```

Reply:
Looks good to me!